// expense-libs/utils/helper.js
const HTTP_CODE = require("expense-libs/utils/constants");

class Helper {

    objResponse = {
        isBase64Encoded: false,
        statusCode: 200,
        headers: {
            "Content-Type": "application/json",
            "Access-Control-Allow-Headers": "*",
            "Access-Control-Allow-Origin": "*",
            "Access-Control-Allow-Methods": "OPTIONS,POST,GET",
            "Access-Control-Allow-Credentials": true
        },
        body: {
            status: 'SUCCESS',
            statusCode: HTTP_CODE.SUCCESS,
            message: '',
            payload: []
        }
    }

    getResponseObject = (status = true, statusCode = HTTP_CODE.SUCCESS, payload = [], message = '') => {
        if (Array.isArray(payload) && !payload.length && !message) {
            this.objResponse.body.message = 'Data not found';
        } else if ((Array.isArray(payload)) || (typeof payload === 'object' && Object.keys(payload).length)) {
            this.objResponse.body.payload = payload;
            this.objResponse.body.message = message !== '' ? message : '';
            this.objResponse.body.statusCode = statusCode !== '' ? statusCode : 200;
            this.objResponse.body.status = status ? 'SUCCESS' : 'FAILURE';
        } else if (payload === null) {
            this.objResponse.body.message = 'Internal server error';
            statusCode = HTTP_CODE.BAD_REQUEST;
            this.objResponse.body.statusCode = HTTP_CODE.BAD_REQUEST;
        }

        return {
            ...this.objResponse,
            statusCode: statusCode,
            body: JSON.stringify(this.objResponse.body)
        };
    }

}

module.exports = Helper;
